hera_cal/lstbin.py

```diff
@@ -175,13 +212,39 @@ def lst_bin(data_list, lst_list, flags_list=None, dlst=None, begin_lst=None, lst
                     if ind not in data[key]:
                         data[key][ind] = []
                         flags[key][ind] = []
+                        nsamples[key][ind] = []
                     # append data ndarray to LST bin
                     data[key][ind].append(d[key][k])
                     # also insert flags if fed
                     if flags_list is None:
                         flags[key][ind].append(np.zeros_like(d[key][k], np.bool))
                     else:
                         flags[key][ind].append(flags_list[i][key][k])
+                    if nsamples_list is None:
+                        nsamples[key][ind].append(np.ones_like(d[key][k], np.int))
+                    else:
+                        nsamples[key][ind].append(nsamples_list[i][key][k])
+        # add in spoofed baselines to keep baselines in different LST files consistent.
+        if bl_list is not None:
+            for antpair in bl_list:
+                for pol in pols:
+                    key = antpair + (pol,)
+                    if key not in data:
+                        if (key[0] != key[1] and utils.reverse_bl(key) not in data) or key[0] == key[1]:
+                            # last part lets us spoof ne and en for autocorrs.
```

Comment:
This makes sure both the `en` and `ne` pols get spoofed for autocorrs. The line you wrote will work fine for cross-correlations but it will only spoof `ne` or `en` (whichever comes first in the polarization list) for an autocorrelation.